### PR TITLE
Use pipenv for packaging

### LINKED ISSUE #5391: Using Pipenv for Packaging
As an alternative to `virtualenv/venv` which is currently used in the project(requirements.txt), we can use pipenv also for easy installation. Also now `pipenv` is officially recommended python package and virtualization manager.
Please refer https://stackoverflow.com/questions/46330327/please-explain-the-usage-of-pipfile-and-pipfile-lock

### PR DESCRIPTION
**Pull request by prabhupant**
Fixes #5391 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
- adds pipenv for packaging
- adds PIpfile and Pipfile.lock
- adds documentation to install Open Server locally using pipenv

#### Changes proposed in this pull request:

- adds pipenv for packaging
- adds PIpfile and Pipfile.lock
- adds documentation to install Open Server locally using pipenv
pipenv

### CODE REVIEW

File: docs/installation/local.md [hunk 1]
```diff
@@ -45,14 +45,42 @@ hint: You may need to upgrade your pip version and install following packages if
...
 ```sh
+pipenv shell
+```
 
+Now after setting up the virtual environment and installing the dependices using either pip or pipenv, type
+
+```sh
 # For linux users
 sudo apt-get install python3-dev
 sudo apt-get install libpq-dev
```
Comment (iamareebjamal) [inline docs/installation/local.md]: Shouldn't this be before pip install?

Comment (prabhupant) [reply docs/installation/local.md]: Oh, yeah. Right :sweat_smile: Fixing it

File: docs/installation/local.md [hunk 2]
```diff
@@ -63,6 +60,38 @@ brew install python@3
 brew install libmagic
 ```
 
+## Using pip and virtualenv
+
+Open a terminal and enter the following commands to setup a virtual environment
+
+```sh
+virtualenv -p python3 venv
+source venv/bin/activate
```
Comment (iamareebjamal) [inline docs/installation/local.md]: Make it .venv

Comment (prabhupant) [reply docs/installation/local.md]: @iamareebjamal done. please review again :+1:

**Comment by prabhupant** [comment]
@iamareebjamal please merge

**Pull request merged**